_posts/2020-10-21-explicit-term-inference-in-scala-3.md

```diff
@@ -0,0 +1,430 @@
+---
+layout: blog-detail
+post-type: blog
+by: Meriam Lachkar, Vincenzo Bazzucchi, Scala Center
+title: Explicit term inference with Scala 3
+---
+
+# Explicit term inference with Scala 3
+
+One of the most striking changes for developers adopting Scala 3 is the introduction
+of new syntaxes to replace the implicit mechanism used in previous Scala versions.
+
+The motivation behind the new syntax is that the same `implicit` keyword was
+used for different purposes and patterns and thus it became a way to express *how* to implement
+patterns. This means that when encountering this dreaded incantation, users need to decipher what
+the intent of the developer was: is this a conversion ? Does this avoid parameter repetition ?
+Is this an extension of a type ? Is this a typeclass ? How do I import this ?
+
+Seeing how pervasive implicits became in libraries and projects, Scala 3 aims at reducing confusion and cognitive load by using new keywords that convey the intent of the developer.
+
+This post briefly introduces the new syntaxes available to Scala 3 programmers by analysing the
+most common use cases and patterns.
```

Comment:
I'd suggest listing those syntaxes here (extension methods, ...), so the reader knows what's coming, and so they'll still get the basic idea even if they don't read the entire blog post. I always suggest this on every blog post: try to make the first few paragraphs a self-contained version of the entire post, so that if somebody only reads the start, they'll still get the important points.